.github/workflows/changelog.yml

```diff
@@ -32,7 +32,8 @@ jobs:
     steps:
       - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5
         with:
-          fetch-depth: 0
+          # Fetch complete history depth only if the PR is not a chore.
+          fetch-depth: ${{ !contains(github.event.pull_request.labels.*.name, 'dependencies') && !contains(github.event.pull_request.labels.*.name, 'Skip Changelog') && !contains(github.event.pull_request.title, '[chore]') && '0' || '1' }}
```

Comment:
Couldn't you move the whole condition 
```
!contains(github.event.pull_request.labels.*.name, 'dependencies') && !contains(github.event.pull_request.labels.*.name, 'Skip Changelog') && !contains(github.event.pull_request.title, '[chore]')
```

to the `changelog` job conditional on line 28 and always fetch with depth 0? I say that because all other steps in the job have the same conditional - this will also allow removing the repetion of the `if` on each step.

Reply:
Not all the steps have the same condition. This step executes no matter what:

```
      - name: Validate ./.chloggen/*.yaml changes
        run: |
          make chlog-validate \
            || { echo "New ./.chloggen/*.yaml file failed validation."; exit 1; }
```

Reply:
Bummer...